File: cp/LC2785-Sort-Vowels-in-a-String2.py
# from builtins import str


class Solution:
    def sortVowels(self, s: str) -> str:
        vowel = "AEIOUaeiou"
        count = {}
        for ch in s:
            if ch in vowel:
                count[ch] = count.get(ch, 0) + 1

        j = 0
        ans = ""
        for ch in s:
            if ch in vowel:
                while count.get(vowel[j], 0) == 0:
                    j += 1
                ans += vowel[j]
                count[vowel[j]] -= 1
            else:
                ans += ch

        return ans


def main() -> None:
    s = input()
    sol = Solution()
    print(sol.sortVowels(s))


if __name__ == "__main__":
    main()

File: cp/LC2785-Sort-Vowels-in-a-String2.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define mod 1000000007

class Solution
{
    bool isVowel(char ch)
    {
        ch = tolower(ch);
        return (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u') ? true : false;
    }

public:
    string sortVowels(string s)
    {
        string temp = "";
        for (auto &ch : s)
        {
            if (isVowel(ch))
                temp += ch;
        }
        sort(temp.begin(), temp.end());
        int j = 0;
        for (auto &ch : s)
        {
            if (isVowel(ch))
            {
                ch = temp[j++];
            }
        }
        return s;
    }
};

int main()
{
    string s;
    cin >> s;

    Solution sol;
    cout << sol.sortVowels(s) << endl;

    return 0;
}